engine/src/Framebuffer.cpp

```diff
@@ -0,0 +1,100 @@
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+
+        CHECKED_GL_CALL(glFramebufferTexture2D, GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0 + i, GL_TEXTURE_2D,
+                        m_colorBuffers[i], 0);
+    }
+
+    unsigned int rboDepth;
+    CHECKED_GL_CALL(glGenRenderbuffers, 1, &rboDepth);
+    CHECKED_GL_CALL(glBindRenderbuffer, GL_RENDERBUFFER, rboDepth);
+    CHECKED_GL_CALL(glRenderbufferStorage, GL_RENDERBUFFER, GL_DEPTH_COMPONENT, m_screenWidth, m_screenHeight);
+    CHECKED_GL_CALL(glFramebufferRenderbuffer, GL_FRAMEBUFFER, GL_DEPTH_ATTACHMENT, GL_RENDERBUFFER, rboDepth);
+
+    unsigned int attachments[2] = {GL_COLOR_ATTACHMENT0, GL_COLOR_ATTACHMENT1};
+    CHECKED_GL_CALL(glDrawBuffers, 2, attachments);
+
+    if (CHECKED_GL_CALL(glCheckFramebufferStatus, GL_FRAMEBUFFER) != GL_FRAMEBUFFER_COMPLETE)
+        std::cout << "ERROR::FRAMEBUFFER:: Framebuffer is not complete!" << std::endl;
+
+    CHECKED_GL_CALL(glBindFramebuffer, GL_FRAMEBUFFER, 0);
+
+    // ping-pong framebuffer for blurring
+    CHECKED_GL_CALL(glGenFramebuffers, 2, m_pingpongFBO);
+    CHECKED_GL_CALL(glGenTextures, 2, m_pingpongColorBuffers);
+    for (unsigned int i = 0; i < 2; i++) {
+        CHECKED_GL_CALL(glBindFramebuffer, GL_FRAMEBUFFER, m_pingpongFBO[i]);
+        CHECKED_GL_CALL(glBindTexture, GL_TEXTURE_2D, m_pingpongColorBuffers[i]);
+        CHECKED_GL_CALL(glTexImage2D, GL_TEXTURE_2D, 0, GL_RGBA16F, m_screenWidth, m_screenHeight, 0, GL_RGBA, GL_FLOAT,
+                        NULL);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+        CHECKED_GL_CALL(glTexParameteri, GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+        CHECKED_GL_CALL(glFramebufferTexture2D, GL_FRAMEBUFFER, GL_COLOR_ATTACHMENT0, GL_TEXTURE_2D,
+                        m_pingpongColorBuffers[i], 0);
+
+        if (CHECKED_GL_CALL(glCheckFramebufferStatus, GL_FRAMEBUFFER) != GL_FRAMEBUFFER_COMPLETE)
+            std::cout << "ERROR::FRAMEBUFFER:: Framebuffer is not complete!" << std::endl;
+    }
+}
+
+void Framebuffer::bind(const unsigned int fbo) {
+    CHECKED_GL_CALL(glBindFramebuffer, GL_FRAMEBUFFER, fbo);
+}
+
+void Framebuffer::bindTexture(const unsigned int t) {
+    CHECKED_GL_CALL(glBindTexture, GL_TEXTURE_2D, t);
+}
+
+void Framebuffer::activeTexture(const unsigned int i) {
+    CHECKED_GL_CALL(glActiveTexture, GL_TEXTURE0 + i);
+}
+
+void Framebuffer::renderQuad() {
+    unsigned int quadVAO = 0;
+    unsigned int quadVBO;
+    if (quadVAO == 0) {
```

Comment:
Ovaj if je uvek tačan, što znači da se u svakom pozivu pravi novi quad, a funkcija `renderQuad` se poziva u ping-pong efektu i trebalo bi da zadrži svoju vrednost.